lib/SetAPI/generic/SetInterfaceV1.py

```diff
@@ -89,12 +131,82 @@ def _check_save_set_params(self: "SetInterfaceV1", params: dict[str, Any]) -> No
             if not params.get(param):
                 raise ValueError(param_required(param))
 
+        # this should not happen with the current API, but best to check
+        if not set_item_name or set_item_name not in SET_ITEM_NAME_TO_SET_TYPE:
+            err_msg = f"invalid set item name: {set_item_name}"
+            raise ValueError(err_msg)
+
+        # bail out early for RNASeqSampleSets
+        if set_item_name == RNASEQ_SAMPLE:
+            return set_item_name
+
+        # convert the set item name into the set type
+        save_search_set = params.get(SAVE_SEARCH_SET, False)
+        if save_search_set and set_item_name == "genome":
+            set_item_name = GENOME_SEARCH
+
+        list_item_type = "elements" if save_search_set else "items"
+        if list_item_type not in params["data"]:
+            raise ValueError(list_required(list_item_type))
+
+        if "description" not in params["data"]:
+            params["data"]["description"] = ""
+
+        # ensure that empty sets are only present in those set types
+        # that allow them
+        if set_item_name not in ALLOWS_EMPTY_SETS and not params["data"].get(
+            list_item_type
+        ):
+            raise ValueError(no_items(SET_ITEM_NAME_TO_SET_TYPE[set_item_name]))
+
+        if list_item_type == "elements":
+            return set_item_name
+
+        # add 'label' fields if not present in data:
+        seen_refs = set()
+        for item in params["data"]["items"]:
+            if "label" not in item:
+                item["label"] = ""
+            if item["ref"] in seen_refs:
+                raise ValueError(no_dupes())
+            seen_refs.add(item["ref"])
+
+        if set_item_name in GENOME_REF_CHECKS:
+            self._check_genome_refs(seen_refs, set_item_name)
+
+        return set_item_name
+
+    def _check_genome_refs(
+        self: "SetInterfaceV1", seen_refs: set[str], set_item_name: str
+    ) -> None:
+        """Ensure that the same genome is referenced by all set members.
```

Comment:
This is a conglomeration of the genome checks from the three modules that do them.